### PR TITLE
samples: matter: Added nRF54H20 support

### PR DESCRIPTION
**Pull request by ArekBalysNordic**
Added support for nRF54H20 DK for Matter Lock and Template samples.

DFU and factory data are not available yet.

Depends on: https://github.com/nrfconnect/sdk-nrf/pull/15686

### CODE REVIEW

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -630,6 +630,13 @@ Matter samples that used :ref:`zephyr:nrf-ieee802154-rpmsg-sample` or :ref:`mult
 
     DFU support for the nRF54L15 PDK is available only for the ``release`` build type.
 
+  * Support for Matter over Thread on the nRF54H20 DK with the ``nrf54h20dk/nrf54h20/cpuapp`` board target to the following Matter samples:
```
Comment (kkasperczyk-no) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:633]:
```suggestion
  * Support for Matter over Thread on the ``nrf54h20dk/nrf54h20/cpuapp`` board target to the following Matter samples:
```

File: samples/matter/lock/boards/nrf54h20dk_nrf54h20_cpuapp.overlay [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+ */
+
+/ {
+	chosen {
+		/* prepare IPC to HCI Radio core */
+		zephyr,bt-hci-ipc = &ipc0;
+	};
+
+	aliases {
+		factory-data = &factory_data;
```
Comment (kkasperczyk-no) [inline samples/matter/lock/boards/nrf54h20dk_nrf54h20_cpuapp.overlay:14]: It is currently not used, right?

File: samples/matter/lock/sysbuild/ipc_radio/boards/nrf54h20dk_nrf54h20_cpurad.overlay [hunk 3]
```diff
@@ -0,0 +1,37 @@
+/* Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+ &dppic130 {
```
Comment (kkasperczyk-no) [inline samples/matter/lock/sysbuild/ipc_radio/boards/nrf54h20dk_nrf54h20_cpurad.overlay:6]: Can we have some comments to this channel configuration? I don't understand anything what is going on here :)

File: samples/matter/template/suit_templates/app_envelope.yaml.jinja2 [hunk 4]
```diff
@@ -0,0 +1,148 @@
+{%- set mpi_application_vendor_name = application['config']['CONFIG_SUIT_MPI_APP_LOCAL_1_VENDOR_NAME']|default('nordicsemi.com') %}
```
Comment (kkasperczyk-no) [inline samples/matter/template/suit_templates/app_envelope.yaml.jinja2:1]: Do this .jinja2 files require documentation? Or they are not something that will be modified by the user, so we can just put them in the sample dir and forget about them?

File: samples/matter/lock/boards/nrf54h20dk_nrf54h20_cpuapp.overlay (hunk 2 above)
Comment (ArekBalysNordic) [reply samples/matter/lock/boards/nrf54h20dk_nrf54h20_cpuapp.overlay:14]: Yes, but I would like to leave it here and repair it ASAP.

File: samples/matter/template/suit_templates/app_envelope.yaml.jinja2 (hunk 4 above)
Comment (ArekBalysNordic) [reply samples/matter/template/suit_templates/app_envelope.yaml.jinja2:1]: We will continue to work on that after the NCS 2.7.0 release, for now, they are generated automatically there.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 1 above)
Comment (ArekBalysNordic) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:633]: Hmm previously we used the same naming for nRF54L15. @wiba-nordic what do you think?

File: samples/matter/lock/sysbuild/ipc_radio/boards/nrf54h20dk_nrf54h20_cpurad.overlay (hunk 3 above)
Comment (ArekBalysNordic) [reply samples/matter/lock/sysbuild/ipc_radio/boards/nrf54h20dk_nrf54h20_cpurad.overlay:6]: Those configurations will be moved to the Zephyr common directory, so we don't need to take care of it. For now, it is a workaround and this file will be removed once it is supported in sdk-zephyr. I will add a note there that it is a workaround.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 1 above)
Comment (wiba-nordic) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:633]: Seems we don't have any consistency on how this is expressed between the different teams in the NCS. We can stick with how it is now (it's the most information), and I'll bring this up with the docs team when we do final release notes review for the next release.

**Review by wiba-nordic — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]

**Pull request merged**